Password optional sign in javascript node guide (#3437)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/nodeexpress/integrationsteps.md

### 1: Create and display sign-in page

Create a sign-in page that captures the user's username.

<div class="common-image-format bordered-image">

> ![Screenshot showing a sign-in page with the username field.](/img/pwd-optional/pwd-optional-sign-in-page.png)

Reviewer: ```suggestion
![Screenshot showing a sign-in page with a username field and a Next button.](/img/pwd-optional/pwd-optional-sign-in-page.png)
```

Author: updated.

</div>

### 2: Submit username

> When the user initiates the sign-in, pass the username to `OktaAuth.idx.authenticate()`.

Reviewer: ```suggestion
When the user submits their username, call `OktaAuth.idx.authenticate()` with the username as a parameter.
```

Author: updated text.

```javascript
  const { username } = req.body;
  const authClient = getAuthClient(req);
  const transaction = await authClient.idx.authenticate({ username });
```

### 3. Verify identity with the email authenticator

`OktaAuth.idx.authenticate()` returns an `IdxTransaction` object indicating that the user needs to verify their identity with the email authenticator challenge.

```json
{
  status: "PENDING",
  nextStep: {
    name: "challenge-authenticator",
    type: "email",
    authenticator: {
      type: "email",
      key: "okta_email",
      displayName: "Email",
    },
  },
}
```

> The email authenticator supports OTP and magic links, and you can integrate both methods into your application. Learn more about integrating the email authenticator challenge by visiting the [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).

Reviewer: ```suggestion
The email authenticator supports OTP and magic links, and you can integrate both methods into your application. For more about integrating the email authenticator, see [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).
```

Author: Updated with slightly differen text.

> ### 4. Complete sign-up

Reviewer: I know th epoint of the article is sign in with email only, but what about the optional authenticators the user has enrolled . Shouldn't there be a step, or at least an acknowledgement in the text, where optional authenticators are listed but skipped, as they are in the Sign-up guide.
```suggestion
### 4. Complete sign-in flow
```

Author: updated.

Author: Add a note since it's out of scope note of use case: In a MFA use case, Find the status, you would have to choose another authenticator and roll through the loop again. And a link to another guide.

Author: updated.

> When the user completes the email authenticator verification, `OktaAuth.idx` should return  `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates a successful sign-in.

Reviewer: ```suggestion
After the email authenticator verifies the user, Identity Engine should return an `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates a successful sign-in.
```

Author: It's the SDK returning SUCCESS, although depending whether they use magic links or OTP the method is different. I've updated the text to say "one of the methods ..." but we can sync about this.

### 1: Create and display sign-in page

Create a sign-in page that captures the user's username.

<div class="half border">

![Screenshot showing a sign-in page with a username field and a Next button.](/img/pwd-optional/pwd-optional-sign-in-page.png)

</div>

### 2: Submit username

> When the user submits their username, call `OktaAuth.idx.authenticate()` passing in the username.

Author: updated.

```javascript
  const { username } = req.body;
  const authClient = getAuthClient(req);
  const transaction = await authClient.idx.authenticate({ username });
```

### 3. Verify identity with the email authenticator

`OktaAuth.idx.authenticate()` returns an `IdxTransaction` object indicating that the user needs to verify their identity with the email authenticator challenge.

```json
{
  status: "PENDING",
  nextStep: {
    name: "challenge-authenticator",
    type: "email",
    authenticator: {
      type: "email",
      key: "okta_email",
      displayName: "Email",
    },
  },
}
```

> The email authenticator supports OTP and magic links, and you can integrate both methods into your application. For more information about integrating the email authenticator challenge, see the [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).

Reviewer 2: ```suggestion
The email authenticator supports OTP and magic links, and you can integrate both methods into your application. See the [Okta email integration guide](/docs/guides/authenticators-okta-email/nodeexpress/main/#integrate-email-challenge-with-magic-links).
```

Author: updated.

### 4. Complete sign-in flow

> When the user completes the email authenticator verification, one of `OktaAuth.idx` methods returns `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates a successful sign-in.

Reviewer 2: ```suggestion
When the user completes the email authenticator verification, one of `OktaAuth.idx` methods returns `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates that the user successfully signed in.
```

Author: updated.

> When the user completes the email authenticator verification, one of `OktaAuth.idx` methods should return `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates a successful sign-in.

Reviewer: ```suggestion
When the user completes the email authenticator verification, one of the `OktaAuth.idx` methods will return an `IdxTransaction.status` of `SUCCESS` along with access and ID tokens, which indicates a successful sign-in.
```

Author: Updated, though I've updated the sentence from past to present tense since we're supposed to avoid future tense.

```json
{
  status: "SUCCESS",
  tokens: {
    accessToken: {
      accessToken: "eyJraWQiOiJLSWdvVHlt...",
      expiresAt: 1656106249,
      tokenType: "Bearer",
    },
    idToken: {
      idToken: "eyJraWQiOiJLSWdvVHltSGlL...",
      expiresAt: 1656106249,
    },
  },
}
```

> > **Note:** In other use cases where additional sign-in authenticators are required, the user would need to choose and verify all required authenticators before `IdxTransaction.status` of `SUCCESS` is returned.

Reviewer 2: ```suggestion
> **Note:** In other use cases where additional sign-in authenticators are required, the user needs to choose and verify all required authenticators before `IdxTransaction.status` of `SUCCESS` is returned.
```

Author: Updated.
